### PR TITLE
Fixes to Locked Pages

### PR DESCRIPTION
**Pull request by balagan73**
#651

### CODE REVIEW

File: web/modules/custom/server_general/src/Routing/LockedPagesRouteSubscriber.php [hunk 1]
```diff
@@ -73,13 +73,11 @@ public function access(AccountInterface $account) {
     $node = $this->routeMatch->getParameter('node');
     // Get the list of bundles that can be restricted.
     $bundles = $this->lockedPagesService->getReferencedBundles();
-    // Return in_array($node->id(), $restricted_entities);.
-    if ($node instanceof NodeInterface && in_array($node->getType(), $bundles)) {
+    // If node is locked, we don't allow accesing the delete page at all.
```
Comment (amitaibu) [inline web/modules/custom/server_general/src/Routing/LockedPagesRouteSubscriber.php:76]: Do we have a test for this? If not,  please add (bump 1h)

Comment (balagan73) [reply web/modules/custom/server_general/src/Routing/LockedPagesRouteSubscriber.php:76]: Yes, we have a test.
https://github.com/Gizra/drupal-starter/blob/846c79ac80eb42ff7dc1ca89dfd471fed09bc019/web/modules/custom/server_general/tests/src/ExistingSite/ServerGeneralNodeLandingPageTest.php#L87

File: web/modules/custom/server_general/src/LockedPages.php [hunk 2]
```diff
@@ -48,7 +49,7 @@ public function __construct(EntityTypeManagerInterface $entity_type_manager, Ent
    * @return \Drupal\Core\Entity\ContentEntityInterface|null
    *   The 'main_settings' config page entity or null if not found.
    */
-  public function getMainSettings() {
+  public function getMainSettings():?ContentEntityInterface {
```
Comment (andreytroeglazov) [inline web/modules/custom/server_general/src/LockedPages.php:52]:
```suggestion
  public function getMainSettings(): ?ContentEntityInterface {
```

File: web/modules/custom/server_general/src/LockedPages.php [hunk 3]
```diff
@@ -62,7 +63,7 @@ public function getMainSettings() {
    * @return bool
    *   Returns TRUE if entity is locked.
    */
-  public function isNodeLocked(NodeInterface $node) {
+  public function isNodeLocked(NodeInterface $node):bool {
```
Comment (andreytroeglazov) [inline web/modules/custom/server_general/src/LockedPages.php:66]:
```suggestion
  public function isNodeLocked(NodeInterface $node): bool {
```

File: web/modules/custom/server_general/src/LockedPages.php [hunk 4]
```diff
@@ -73,7 +74,7 @@ public function isNodeLocked(NodeInterface $node) {
    * @return array
    *   List of ids.
    */
-  protected function getRestrictedNodes() {
+  protected function getRestrictedNodes():array {
```
Comment (andreytroeglazov) [inline web/modules/custom/server_general/src/LockedPages.php:77]:
```suggestion
  protected function getRestrictedNodes(): array {
```

File: web/modules/custom/server_general/src/LockedPages.php [hunk 5]
```diff
@@ -91,7 +92,7 @@ protected function getRestrictedNodes() {
    * @return array
    *   An array of bundle type machine names.
    */
-  public function getReferencedBundles() {
+  public function getReferencedBundles():array {
```
Comment (andreytroeglazov) [inline web/modules/custom/server_general/src/LockedPages.php:95]:
```suggestion
  public function getReferencedBundles(): array {
```

**Review by andreytroeglazov — APPROVED** [review]

**Pull request merged**